manual syncpack after rc.14 release

spmonahan opened the pull request:
Bumps dependencies after a release. This is manual fix for a known issue.

fabricteam commented:
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/11b73e5cce830963db3b34201c7da2314cab3ef7'>11b73e5cce830963db3b34201c7da2314cab3ef7</a></sub>
<div id="bundleSizeComment9423"/>

spmonahan commented:
> Syncpack author here 👋 Is the need for this something that can be fixed within syncpack? Or something else related to this repo? Thanks.

It's both. One part is that our build tooling doesn't support bumping package versions in certain cases. #22559 describes this issue and AFAIK it has nothing to do with syncpack.

To work around this issue with our build tool we manually run syncpack and make a PR like this one. However, I've not been able to use `syncpack fix-mismatches` as it incorrectly downgrades some dependencies. Rather I used `syncpack list-mismatches` to identify files to look at and manually bump versions.

Here are my repro steps:

1. Clone this repo.
2. Checkout the last commit before this pr: `git checkout -b syncpack-repro 11b73e5cce830963db3b34201c7da2314cab3ef7`
3. Install deps: `yarn install`
4. List mismatches: `yarn syncpack list-mismatches`
    1. Get 
        ```
        - ^9.0.0-rc.13 in dependencies of @fluentui/public-docsite-v9
        - ^9.0.0-rc.14 in dependencies of @fluentui/ts-minbar-test-react-components
        - ^9.0.0-rc.14 in dependencies of @fluentui/theme-designer
        - ^9.0.0-rc.14 in devDependencies of @fluentui/react-portal-compat 
        ```
        I only expect `@fluentui/public-docsite-v9` to be in that list as it's at rc.13 (previous version) while everything else is at rc.14 (correct, current version)
5. Fix mismatches: `yarn syncpack fix-mismatches`
    1.  Changes in git
    ```
    modified:   apps/ts-minbar-test-react-components/package.json
    modified:   packages/react-components/react-portal-compat/package.json
    modified:   packages/react-components/theme-designer/package.json
    ```
    All of these files have been downgraded from rc.14 to rc.13. The `@fluentui/public-docsite-v9` that I expected to get upgraded from rc.13 to rc.14 has not changed.
